packages/form/MIGRATION.md

```diff
@@ -0,0 +1,135 @@
+# Migrating from `availity-reactstrap-validation` to `@availity/form`
+
+The primary difference in developing with `availity-reactstrap-validation` and `@availity/form` is validation. With `availity-reactstrap-validation`, we define validation schemas on the input components in the form with the `validate` prop, like so:
+
+```jsx
+import { AvInput, AvForm } from 'availity-reactstrap-validation';
+
+// ...
+<AvForm>
+  <AvInput
+    name="myInput"
+    type="text"
+    validate={{
+      required: {
+        value: true,
+        errorMessage: 'This field is required.',
+      },
+    }}
+  />
+</AvForm>
+// ...
+```
+
+Note: `availity-reactstrap-validation` also has a `required` prop that acts as a shorthand for required validation using the `validate` prop. The code snippet above is equivalent to:
+
+```jsx
+import { AvInput, AvForm } from 'availity-reactstrap-validation';
+
+// ...
+<AvForm>
+  <AvInput
+    name="myInput"
+    type="text"
+    required
+  />
+</AvForm>
+// ...
+```
+
+With `@availity/form`, we use `yup` and `@availity/yup` to define the validation schema for all of the inputs in the form on the `<Form />` component with the `validationSchema` prop from [formik](https://jaredpalmer.com/formik/docs/guides/validation#validationschema), like so:
+
+```jsx
+import { Input, Form } from '@availity/form';
+
+// ...
+<Form
+  initialValues={{ myInput: '' }}
+  validationSchema={yup.object().shape({ myInput: yup.string().isRequired(true, 'This field is required.') })}
+>
+  <Input name="myInput" type="text" />
```

Comment:
Should this be `<Input name="myInput" type="text" required />` (making it kinda the same as the old way) for a11y? I'm not sure if the validationSchema is automatically added the HTML5 equivalent or not, but if not, we should it with a11y as a11y shouldn't be an afterthought and the examples should show best practices.
It also helps keep the examples as equivalent.

Similar for the ones before which involve pattern and required.

Reply:
I like the idea of automatically adding the `required` attribute to the input if it is required from `validationSchema`. I created the following issue: https://github.com/Availity/availity-react/issues/245

Reply:
I like the idea. I created #245